### PR TITLE
Document minitest-queue usage

### PR DESCRIPTION
**Pull request by casperisfine**
Pretty much all is in the title. The documentation might still be a little light but it's better than nothing I guess...

I think after this PR we can cut a new release.

### CODE REVIEW

File: ruby/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+Or install it yourself as:
+
+    $ gem install ci-queue
+
+## Usage
+
+### Supported CI providers
+
+`ci-queue` automatically infer most of it's configuration if ran on one of the following CI providers:
```
Comment (wvanbergen) [inline ruby/README.md]: Small `infers` & `its configuration`

File: ruby/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  - Travis
+
+If you are using another CI system, please refer to the command usage message.
+
+### Minitest
+
+Assuming you use one of the supported CI providers, the command can be as simple as:
+
+```bash
+minitest-queue --url redis://example.com run -Itest test/**/*_test.rb
```
Comment (wvanbergen) [inline ruby/README.md]: I find the meaning of `--url` a bit vague. Maybe instead of using `--url` we should use `--queue` ? Not sure.

File: ruby/lib/ci/queue/configuration.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@ def from_env(env)
       end
 
       def initialize(
-        timeout: 10, build_id: nil, worker_id: nil, max_requeues: 0, requeue_tolerance: 0,
+        timeout: 30, build_id: nil, worker_id: nil, max_requeues: 0, requeue_tolerance: 0,
```
Comment (wvanbergen) [inline ruby/lib/ci/queue/configuration.rb]: Why this change?

File: ruby/README.md (hunk 2 above)
Comment (wvanbergen) [reply ruby/README.md]: Or, `--orchestrator` or `--coordinator`

File: ruby/lib/ci/queue/configuration.rb (hunk 3 above)
Comment (casperisfine) [reply ruby/lib/ci/queue/configuration.rb]: Because I figured 10 seconds is very low, and people trying out this gem might be surprised by the behavior. 30 seemed fair.

File: ruby/README.md (hunk 1 above)
Comment (casperisfine) [reply ruby/README.md]: Fixed.

File: ruby/README.md (hunk 2 above)
Comment (casperisfine) [reply ruby/README.md]: Changed it for `--queue`

File: ruby/lib/minitest/queue/runner.rb [hunk 4]
```diff
@@ -76,76 +80,141 @@ def report_command
...
+          EOS
+          opts.separator ""
+          opts.on('--worker WORKER_ID', *help) do |worker_id|
             queue_config.worker_id = worker_id
           end
 
+          help = split_heredoc(<<-EOS)
+            Defines how many time a single test can be requeued.
+            Defaults to 0.
+          EOS
```
Comment (wvanbergen) [inline ruby/lib/minitest/queue/runner.rb]: Not sure about the minimum Ruby version you had in mind, but if you'd use `<<~EOS` there's no need for stripping the string, e.g. `opts.on('--requeue-tolerance RATIO', *help.lines)` without `split_heredoc`.

File: README.md [hunk 5]
```diff
@@ -12,37 +12,36 @@ There are algorithms available to balance perfectly your workers, but in practic
 
 Another advantage is that if you lose workers along the way, using a queue the other workers can pick up the job, making you resilient to failures.
 
-## Installation
+## How does it work?
 
-Add this line to your application's Gemfile:
+Every workers first participate in a leader election, the elected leader will then populate the queue with the list of tests to run.
```
Comment (DazWorrall) [inline README.md]: Suggest `Every workers first participate` -> `Each worker first participates`

File: README.md [hunk 6]
```diff
@@ -12,37 +12,36 @@ There are algorithms available to balance perfectly your workers, but in practic
...
+Once the queue is initialized, all workers including the leader will reserve tests and work them on until the queue is empty.
 
-```ruby
-gem 'ci-queue'
-```
+If a worker were to die, it's reserved work load will be put back into the queue to be picked up by it's surviving siblings.
```
Comment (DazWorrall) [inline README.md]: Grammar suggestion (twice in this sentence): `it's` -> `its`

File: ruby/lib/minitest/queue/runner.rb (hunk 4 above)
Comment (casperisfine) [reply ruby/lib/minitest/queue/runner.rb]: Lines have to be passed as an array of strings, otherwise the formatting is messed up. So it doesn't change much.

Also it would have been two small of an improvement for requiring 2.3. That's why I didn't use `<<~`

File: README.md [hunk 7]
```diff
@@ -12,37 +12,36 @@ There are algorithms available to balance perfectly your workers, but in practic
...
 
-And then execute:
+Additionally, a separate process can be started to centralize the error reporting, that process will wait for the queue to be empty before exiting.
 
-    $ bundle
+## What are requeues?
 
-Or install it yourself as:
+When working on big test suites, it's not uncommon for a few tests to fail intermittently, either because they are inherently flaky,
+or either because they are sensible to other tests modifying some global state (leaky tests).
```
Comment (DazWorrall) [inline README.md]: `or either because they are sensible to` -> `or because they are sensitive to`

**Comment by casperisfine** [comment]
Thanks @DazWorrall.

**Comment by casperisfine** [comment]
I'll cut a release in a bit.
